### ISSUE TITLE
How to configure multiple worker threads in CN

### ISSUE THREAD

**Issue by SeKwonLee** [WukLab/pDPM#4]
I am trying to run experiments using multiple worker threads in CN. However, the current implementation looks just returning created threads while only allowing a single worker to run due to code lines below in the `mitsume_benchmark_latency` function. Is there a specific reason for this implementation?

```
  if (thread_metadata->thread_id != 0)
    return NULL;
```

To enable multiple worker threads, I have tried several things. I removed those lines and used `thread_metadata->thread_id` in assigning input/output buffers like below (in the `mitsume_benchmark_latency` function). Furthermore, I changed `mitsume_tool_open`, `mitsume_tool_read`, and `mitsume_tool_write` to receive `thread_metadata->thread_id` as their `coro_id`.

```
  write = local_inf->user_input_space[thread_metadata->thread_id];
  read = local_inf->user_output_space[thread_metadata->thread_id];
  if (client_id != 0)
    return NULL;

  if (stick_this_thread_to_core(thread_metadata->thread_id)) {
    printf("set affinity fail\n");
    return NULL;
  }
```

However, after the modifications, I got errors below during runtime from CN. Could you give me any suggestions to correctly use multiple worker threads?

```
init: /usr/include/boost/coroutine/detail/symmetric_coroutine_yield.hpp:290: boost::coroutines::detail::symmetric_coroutine_yield<void>& boost::coroutines::detail::symmetric_coroutine_yield<void>::operator()(Coro&, typename boost::enable_if<boost::is_same<typename Coro::value_type, void>, boost::coroutines::detail::symmetric_coroutine_yield<void>::dummy*>::type) [with Coro = boost::coroutines::detail::symmetric_coroutine_call<void>; typename boost::enable_if<boost::is_same<typename Coro::value_type, void>, boost::coroutines::detail::symmetric_coroutine_yield<void>::dummy*>::type = boost::coroutines::detail::symmetric_coroutine_yield<void>::dummy*]: Assertion `other' failed.
init: /usr/include/boost/coroutine/detail/symmetric_coroutine_yield.hpp:290: boost::coroutines::detail::symmetric_coroutine_yield<void>& boost::coroutines::detail::symmetric_coroutine_yield<void>::operator()(Coro&, typename boost::enable_if<boost::is_same<typename Coro::value_type, void>, boost::coroutines::detail::symmetric_coroutine_yield<void>::dummy*>::type) [with Coro = boost::coroutines::detail::symmetric_coroutine_call<void>; typename boost::enable_if<boost::is_same<typename Coro::value_type, void>, boost::coroutines::detail::symmetric_coroutine_yield<void>::dummy*>::type = boost::coroutines::detail::symmetric_coroutine_yield<void>::dummy*]: Assertion `other' failed.
init: /usr/include/boost/coroutine/detail/symmetric_coroutine_yield.hpp:290: boost::coroutines::detail::symmetric_coroutine_yield<void>& boost::coroutines::detail::symmetric_coroutine_yield<void>::operator()(Coro&, typename boost::enable_if<boost::is_same<typename Coro::value_type, void>, boost::coroutines::detail::symmetric_coroutine_yield<void>::dummy*>::type) [with Coro = boost::coroutines::detail::symmetric_coroutine_call<void>; typename boost::enable_if<boost::is_same<typename Coro::value_type, void>, boost::coroutines::detail::symmetric_coroutine_yield<void>::dummy*>::type = boost::coroutines::detail::symmetric_coroutine_yield<void>::dummy*]: Assertion `other' failed.
```

**Comment by lastweek** [comment]
Hi @SeKwonLee ,

Thanks for bringing this to our attention. I think I have removed some useful sample code that demonstrates this capability. I added them back in the latest commit.

The `mitsume_benchmark_latency` was meant to _only_ test end-to-end single-thread latency, therefore it is written in this way.

The `mitsume_benchmark_ycsb` is a multi-threaded implementation, where `MITSUME_BENCHMARK_THREAD_NUM` number of threads will be created. I think the key is that we should only run mitsume_tool_open once, per: https://github.com/WukLab/pDPM/blob/master/clover/mitsume_benchmark.cc#L260
